cmake/modules/CheckCompiler.cmake

```diff
@@ -97,43 +97,42 @@ endif()
 include(CheckCXXCompilerFlag)
 include(CheckCCompilerFlag)
 
-#---Check for cxx11 option------------------------------------------------------------
-if(cxx11 AND cxx14)
-  message(STATUS "c++11 mode requested but superseded by request for c++14 mode")
-  set(cxx11 OFF CACHE BOOL "" FORCE)
-endif()
-if((cxx11 OR cxx14) AND cxx17)
-  message(STATUS "c++11 or c++14 mode requested but superseded by request for c++17 mode")
-  set(cxx11 OFF CACHE BOOL "" FORCE)
-  set(cxx14 OFF CACHE BOOL "" FORCE)
-endif()
-if(cxx11)
-  CHECK_CXX_COMPILER_FLAG("-std=c++11" HAS_CXX11)
-  if(NOT HAS_CXX11)
-    message(STATUS "Current compiler does not suppport -std=c++11 option. Switching OFF cxx11 option")
-    set(cxx11 OFF CACHE BOOL "" FORCE)
-  endif()
-endif()
-if(cxx14)
-  CHECK_CXX_COMPILER_FLAG("-std=c++14" HAS_CXX14)
-  if(NOT HAS_CXX14)
-    message(STATUS "Current compiler does not suppport -std=c++14 option. Switching OFF cxx14 option")
-    set(cxx14 OFF CACHE BOOL "" FORCE)
+#---C++ standard----------------------------------------------------------------------
+
+set(CMAKE_CXX_STANDARD 11 CACHE STRING "")
```

Comment:
That seems surprising. Doesn't that override a user setting? Can we not simply take the compiler default as our default?

Reply:
No, the command line will override this, since this is setting in the cache. It only overrides if `FORCE` is used.